src/logql-query.ts

```diff
@@ -0,0 +1,48 @@
+import { notEmptyString, trim } from './value-utils';
+
+export class LogQLQuery {
+  selectors: Array<string> = [];
+  pipeline: Array<string> = [];
+
+  constructor(query: string) {
+    const parts = query.match(/(.*){([^}]*)}(.*)/);
+    this.selectors =
+      parts && parts.length > 1
+        ? parts[2].split(',').filter(notEmptyString).map(trim)
```

Comment:
I think that if the regex matches, you are guaranteed to always get 4 entries in `parts` (the whole string in `parts[0]` then 3 entries for each of the brackets, even if they are empty strings). So the `parts.length > 1` and `parts.length > 2` tests don't actually do anything useful.